src/SubscribeMe/Subscriber/SubscriberInterface.php

```diff
@@ -20,8 +22,17 @@ public function setContactListId(?string $contactListId): SubscriberInterface;
      * @param string      $email
      * @param array       $options
      * @param UserConsent[] $userConsents
-     *
-     * @return mixed Contact ID if succeeded or false
+     * @return bool|int Contact ID if succeeded or false
+     * @throws JsonException
      */
-    public function subscribe(string $email, array $options, array $userConsents = []);
+    public function subscribe(string $email, array $options, array $userConsents = []): bool|int;
+
+    /**
+     * @param array<EmailAddress> $emails
+     * @param string|int $emailTemplateId
+     * @param array $variables
```

Comment:
```suggestion
     * @param array<string, string|int|bool|array<string|int|bool>> $variables
```

You should require an array form here. To make it generic for all subscribers. Then transform variables to suit every Subscribers